libs/datasets/weekly_hospitalizations.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+import pandas as pd
+
+from libs.datasets import timeseries
+from datapublic.common_fields import CommonFields
+from libs.datasets.dataset_utils import AggregationLevel
+
+MultiRegionDataset = timeseries.MultiRegionDataset
+
+
+def _rolling_sum_7day(daily: pd.Series):
+    assert CommonFields.DATE in daily.index.names
+    return daily.rolling(7).sum()
+
+
+def calculate_weekly_column_from_daily(
+    dataset_in: MultiRegionDataset,
+    level_to_replace: AggregationLevel,
+    field_in: CommonFields,
+    field_out: CommonFields,
+) -> MultiRegionDataset:
+    """Calculate weekly occurrences from daily data using a rolling 7-day sum.
+
+    Args:
+        dataset_in: The dataset to update.
+        level_to_replace: Aggregation level to update data for.
+        field_in: Column to sum into weekly data, should have units of new daily occurrences (e.g. new daily hospital admissions).
+        field_out: Column to sum data into. Result will have units of new weekly occurrences (e.g. new weekly hospital admissions).  
+    """
+
+    state_ds = dataset_in.get_subset(aggregation_level=level_to_replace)
+    state_ts = state_ds.timeseries_bucketed.copy()  # copy to avoid SettingWithCopy warning.
+
+    # If there's no data for field_in then just return the original dataset.
+    if not state_ts[field_in].first_valid_index():
+        return dataset_in
+
+    # Make sure we do not have any pre-existing weekly data before overwriting the field_out column.
+    assert not state_ts[field_out].any()
```

Comment:
We could instead make this non-breaking and just use existing data if it exists instead of calculating it, but we shouldn't ever (at the moment) have any raw weekly data at the state level, so if this gets thrown then that probably means something is wrong.